test/integration/menu-bar.test.js

```diff
@@ -0,0 +1,60 @@
+import path from 'path';
+import SeleniumHelper from '../helpers/selenium-helper';
+
+const {
+    clickXpath,
+    findByXpath,
+    getDriver,
+    loadUri
+} = new SeleniumHelper();
+
+const uri = path.resolve(__dirname, '../../build/index.html');
+
+let driver;
+
+describe('Menu bar settings', () => {
+    beforeAll(() => {
+        driver = getDriver();
+    });
+
+    afterAll(async () => {
+        await driver.quit();
+    });
+
+    test('File->New should be enabled', async () => {
+        await loadUri(uri);
+        await clickXpath('//button[@title="Try It"]');
+        await clickXpath(
+            '//div[contains(@class, "menu-bar_menu-bar-item") and ' +
+            'contains(@class, "menu-bar_hoverable")][span[text()="File"]]'
+        );
+        await findByXpath('//*[li[span[text()="New"]] and not(@data-tip="tooltip")]');
```

Comment:
So the approach i used was to make strict xpaths that should all be found, where if the capability flag was handled wrong, the tooltip status wouldn't match and the findByXpath will timeout. 

This works, but it seems not ideal. Do you have any suggestions for a better way? I couldn't figure out how to make sense of the promise object that findByXpath returns, but maybe I could somehow use then/catch?

Reply:
Check out what `findByXpath` does — it doesn't actually return anything you can do something with, it just waits. It should probably be renamed `waitUntilXpathFound`.  If you want to do something with the elements, you should use `driver.findElements`.  This worked when I tried it:

```js
        const el = await driver.findElements(By.xpath('//*[li[span[text()="New"]] and not(@data-tip="tooltip")]'));
        expect(el.length).toEqual(1);
```